### DISCUSSION TITLE
If I start the spindle, ALARM:13 is displayed

### DISCUSSION THREAD

**Discussion by andy0983** [Paciente8159/uCNC#620] [Config and issues] [answered]
**Describe the bug**
If I start the spindle, ALARM is displayed, but not always
Examples 1:
M3 S12000
ok
[MSG:Reset to continue]
[MSG:Reset to continue]
ok
[MSG:Reset to continue]
Examples 2:
M3 S12000
ok
ALARM:13
[MSG:'$H'|'$X' to unlock]

Describe your hardware:
1. What's your board
uCNC 1.8.6 -master
Arduino Mega  -  Ramps 1.4

2. Paste your "Grbl/µCNC settings"

#ifndef BOADMAP_OVERRIDES_H
#define BOADMAP_OVERRIDES_H
#ifdef __cplusplus
extern "C"
{
#endif

#include "boardmap_reset.h"
#define MCU MCU_AVR
#define KINEMATIC KINEMATIC_CARTESIAN
#define AXIS_COUNT 3
#define TOOL_COUNT 1
#define BAUDRATE 115200
#define S_CURVE_ACCELERATION_LEVEL 0
#define BOARD BOARD_RAMPS14
#define BOARD_NAME "RAMPS 1.4"
#define UART_PORT 0
#define UART2_PORT 2
#define ITP_TIMER 1
#define RTC_TIMER 0
#define ONESHOT_TIMER 4
#define STEP0_BIT 0
#define STEP0_PORT F
#define STEP1_BIT 6
#define STEP1_PORT F
#define STEP2_BIT 3
#define STEP2_PORT L
#define STEP3_BIT 1
#define STEP3_PORT C
#define STEP4_BIT 4
#define STEP4_PORT A
#define DIR0_BIT 1
#define DIR0_PORT F
#define DIR1_BIT 7
#define DIR1_PORT F
#define DIR2_BIT 1
#define DIR2_PORT L

#define DIR3_BIT 3
#define DIR3_PORT C
#define DIR4_BIT 6
#define DIR4_PORT A
#define STEP0_EN_BIT 7
#define STEP0_EN_PORT D
#define STEP1_EN_BIT 2
#define STEP1_EN_PORT F
#define STEP2_EN_BIT 0
#define STEP2_EN_PORT K

#define STEP3_EN_BIT 7
#define STEP3_EN_PORT C
#define STEP4_EN_BIT 2
#define STEP4_EN_PORT A

#define PWM0_BIT 5
#define PWM0_PORT G
#define PWM0_CHANNEL B
#define PWM0_TIMER 0
#define PWM1_BIT 6
#define PWM1_PORT H
#define PWM1_CHANNEL B
#define PWM1_TIMER 2
#define PWM2_BIT 5
#define PWM2_PORT H
#define PWM2_CHANNEL C
#define PWM2_TIMER 4
#define DOUT0_BIT 4
#define DOUT0_PORT B
#define DOUT1_BIT 5
#define DOUT1_PORT K
#define DOUT2_BIT 7
#define DOUT2_PORT L
#define DOUT3_BIT 5
#define DOUT3_PORT L
#define DOUT29_BIT 5
#define DOUT29_PORT A
#define DOUT30_BIT 7
#define DOUT30_PORT A
#define DOUT31_BIT 7
#define DOUT31_PORT B
#define LIMIT_X_BIT 1
#define LIMIT_X_PORT J
#define LIMIT_X_ISR 1
#define LIMIT_Y_BIT 0
#define LIMIT_Y_PORT J
#define LIMIT_Y_ISR 1
#define LIMIT_Z_BIT 2
#define LIMIT_Z_PORT D
#define LIMIT_Z_ISR -3
#define LIMIT_Y2_BIT 3
#define LIMIT_Y2_PORT D
#define LIMIT_Y2_ISR -4
#define PROBE_BIT 4
#define PROBE_PORT E
#define PROBE_ISR -5
#define ESTOP_BIT 1
#define ESTOP_PORT K
#define ESTOP_ISR 2
#define SAFETY_DOOR_BIT 4
#define SAFETY_DOOR_PORT K
#define SAFETY_DOOR_ISR 2
#define FHOLD_BIT 2
#define FHOLD_PORT K
#define FHOLD_ISR 2
#define CS_RES_BIT 1
#define CS_RES_PORT K
#define CS_RES_ISR 2
#define DIN0_BIT 5
#define DIN0_PORT E
#define DIN0_PULLUP
#define DIN0_ISR -6
#define DIN16_PULLUP
#define DIN17_PULLUP
#define DIN18_PULLUP
#define DIN19_BIT 0
#define DIN19_PORT L
#define DIN19_PULLUP
#define DIN29_BIT 3
#define DIN29_PORT A
#define TX_BIT 1
#define TX_PORT E
#define RX_BIT 0
#define RX_PORT E
#define SPI_CLK_BIT 1
#define SPI_CLK_PORT B
#define SPI_SDI_BIT 3
#define SPI_SDI_PORT B
#define SPI_SDO_BIT 2
#define SPI_SDO_PORT B
#define SPI_CS_BIT 0
#define SPI_CS_PORT B
#define I2C_CLK_BIT 0
#define I2C_CLK_PORT D
#define I2C_DATA_BIT 1
#define I2C_DATA_PORT D
#define TX2_BIT 1
#define TX2_PORT H
#define RX2_BIT 0
#define RX2_PORT H

#ifndef CNC_HAL_OVERRIDES_H
#define CNC_HAL_OVERRIDES_H
#ifdef __cplusplus
extern "C"
{
#endif

#include "cnc_hal_reset.h"
#define ENABLE_COOLANT
#define ENABLE_SKEW_COMPENSATION
#define ENABLE_BACKLASH_COMPENSATION
#define ENABLE_EXTRA_SYSTEM_CMDS
#define FORCE_SOFT_POLLING
#define LIMIT_X_PULLUP_ENABLE
#define LIMIT_Y_PULLUP_ENABLE
#define LIMIT_Z_PULLUP_ENABLE
#define LIMIT_X2_PULLUP_ENABLE
#define LIMIT_Y2_PULLUP_ENABLE
#define LIMIT_Z2_PULLUP_ENABLE
#define PROBE_PULLUP_ENABLE
#define ENABLE_MULTI_STEPPER_AXIS
#define LINACT0_IO_MASK 1
#define LINACT1_IO_MASK 10
#define LINACT2_IO_MASK 4
#define ENABLE_AXIS_AUTOLEVEL
#define LIMIT_X_IO_MASK 1
#define LIMIT_Y_IO_MASK 2
#define LIMIT_Z_IO_MASK 4
#define LIMIT_Y2_IO_MASK 8
#define TOOL1 spindle_pwm
#define SPINDLE_PWM PWM0
#define SPINDLE_PWM_DIR DOUT0
#define SPINDLE_PWM_COOLANT_FLOOD DOUT2
#define SPINDLE_PWM_COOLANT_MIST DOUT3
#define SOFT_SPI_CLK DOUT30
#define SOFT_SPI_SDO DOUT29
#define SOFT_SPI_SDI DIN29
#define IC74HC595_COUNT 0
#define ENCODERS 0
#define ENABLE_MAIN_LOOP_MODULES
#define ENABLE_IO_MODULES
#define ENABLE_PARSER_MODULES
#define ENABLE_MOTION_CONTROL_MODULES
#define ENABLE_RT_SYNC_MOTIONS
//#define G5_MAX_SEGMENT_LENGTH 1
//#define LOAD_MODULES_OVERRIDE() ({LOAD_MODULE(g5);LOAD_MODULE(g7_g8);LOAD_MODULE(g33);LOAD_MODULE(m17_m18);LOAD_MODULE(m42);LOAD_MODULE(m62_m65);LOAD_MODULE(m67_m68);})

#ifdef __cplusplus
}
#endif
#endif

#ifndef CONFIG_AT_H
#define CONFIG_AT_H

#ifdef __cplusplus
extern "C"
{
#endif

/*
We copy to file cnc_hal_config_helper.h
#include "config_AT.h"
*/

/ * $0 * /								
/ * $1 * / 								
/ * $2 * /  #define DEFAULT_STEP_INV_MASK 0             
/ * $3 * /  #define DEFAULT_DIR_INV_MASK 0               
/ * $4 * /  #define DEFAULT_STEP_ENA_INV 0                
/ * $5 * /  #define DEFAULT_LIMIT_INV_MASK 0             
/ * $6 * /  #define DEFAULT_PROBE_INV_MASK 0             
/ * $7 * /  #define DEFAULT_CONTROL_INV_MASK 0          
/ * $8 * /                                              
/ * $9 * /                                              
/ * $10 * / #define DEFAULT_STATUS_MASK 1               
/ * $11 * / #define DEFAULT_G64_FACTOR 0.01     
/ * $12 * / #define DEFAULT_ARC_TOLERANCE 0.02		
/ * $13 * / #define DEFAULT_REPORT_INCHES 0		
/ * $20 * / #define DEFAULT_SOFT_LIMITS_ENABLED 0	
/ * $21 * / #define DEFAULT_HARD_LIMITS_ENABLED 1			
/ * $22 * / #define DEFAULT_HOMING_ENABLED 1			
/ * $23 * / #define DEFAULT_HOMING_DIR_INV_MASK 6		
/ * $24 * / #define DEFAULT_HOMING_SLOW 50			
/ * $25 * / #define DEFAULT_HOMING_FAST 600				
/ * $26 * / #define DEFAULT_DEBOUNCE_MS 200			
/ * $27 * / #define DEFAULT_HOMING_OFFSET 5				
/ * $30 * / #define DEFAULT_SPINDLE_MAX_RPM 24000		
/ * $31 * / #define DEFAULT_SPINDLE_MIN_RPM 0 		
/ * $32 * / #define DEFAULT_ENABLE_LASER_PPI 0		
/ * $33 * / #define DEFAULT_LASER_PPI 254					
/ * $34 * / #define DEFAULT_LASER_PPI_USWIDTH	 1500	
/ * $35 * /	
/ * $36 * /	
/ * $37 * /											
/ * $38 * /												
/ * $39 * /												
/ * $80 * / 												
/ * $81 * /												
/ * $100* / #define DEFAULT_STEP_PER_MM_PER_X 50			
/ * $101* / #define DEFAULT_STEP_PER_MM_PER_Y 50          
/ * $102* / #define DEFAULT_STEP_PER_MM_PER_Z 50			
 //#define DEFAULT_STEP_PER_MM 0
#define DEFAULT_STEP_PER_MM_PER_AXIS {DEFAULT_STEP_PER_MM_PER_X, DEFAULT_STEP_PER_MM_PER_Y, DEFAULT_STEP_PER_MM_PER_Z}
/ * $110* / #define DEFAULT_MAX_FEED_PER_X 3000			
/ * $111* / #define DEFAULT_MAX_FEED_PER_Y 3000			
/ * $112* / #define DEFAULT_MAX_FEED_PER_Z 500			
 //#define DEFAULT_MAX_FEED 0
 #define DEFAULT_MAX_FEED_PER_AXIS {DEFAULT_MAX_FEED_PER_X, DEFAULT_MAX_FEED_PER_Y, DEFAULT_MAX_FEED_PER_Z}
/ * $120* / #define DEFAULT_ACCEL_PER_X 50				
/ * $121* / #define DEFAULT_ACCEL_PER_Y 50				
/ * $122* / #define DEFAULT_ACCEL_PER_Z 20				
//#define DEFAULT_ACCEL 0
#define DEFAULT_ACCEL_PER_AXIS {DEFAULT_ACCEL_PER_X , DEFAULT_ACCEL_PER_Y, DEFAULT_ACCEL_PER_Z}
/ * $130* / #define DEFAULT_MAX_DIST_PER_X 650			
/ * $131* / #define DEFAULT_MAX_DIST_PER_Y 650			
/ * $132* / #define DEFAULT_MAX_DIST_PER_Z 100			
//#define DEFAULT_MAX_DIST 0
 #define DEFAULT_MAX_DIST_PER_AXIS {DEFAULT_MAX_DIST_PER_X , DEFAULT_MAX_DIST_PER_Y, DEFAULT_MAX_DIST_PER_Z}
/ * $140* /												
/ * $141* /												
/ * $142* /												h

//cnc_config.h
#define ENABLE_SETTINGS_MODULES
#define GCODE_PROCESS_LINE_NUMBERS
#define ENABLE_RT_SYNC_MOTIONS
#define ENABLE_MAIN_LOOP_MODULES
#define ENABLE_IO_MODULES
#define ENABLE_PARSER_MODULES 
#define ENABLE_MOTION_CONTROL_MODULES
#define ENABLE_EXTRA_SYSTEM_CMDS

//cnc_hal_config.h
#define ESTOP_PULLUP_ENABLE
#define SAFETY_DOOR_PULLUP_ENABLE
#define FHOLD_PULLUP_ENABLE
#define CS_RES_PULLUP_ENABLE
#define LIMIT_X2_DISABLE
#define LIMIT_Z2_DISABLE
#define LIMIT_A_DISABLE
#define LIMIT_B_DISABLE
#define LIMIT_C_DISABLE
#define DISABLE_A_HOMING
#define DISABLE_B_HOMING
#define DISABLE_C_HOMING
#define SYSTEM_MENU_MAX_STR_LEN 32	

#ifdef __cplusplus
}
#endif

#endif

**Comment by Paciente8159** [comment, marked as answer]
If this only happens when you run the spindle...most probably it's not a bug but a noise issue.
If you can run GCODE ok and nothing goes wrong and when you turn the spindle on it trips an alarm now and then you have  a noise issue.

You can confirm this by running the same GCODE with the spindle unpowered (but still using M3, M4 and M5 commands) and nothing trips.

You can solve this with hardware. I recommend using some optocouplers (like the inexpensive PC817) and work with 2 voltage levels.  5V on the Arduino side (near the board) and 12V or 24V (or whatever power supply you are using on the other side (on the limits wires).

Here is an example:
https://planet-cnc.com/wp-content/uploads/2016/02/switch.png

I've also had this issue with my Spindle+VFD. The optocouplers solved it.

**EDIT**
Actually I use a circuit like the one above but use N.C. switches. That allows me to detect also unconnected limits or faulty wires.

**Comment by andy0983** [comment]
You were right, there are interferences with the Spindle+VFD and the relay that turns on the spindle cooling
Thank you for your help
